samples/kafkaBasic/Producer/ProducerWorker.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Confluent.Kafka;
+
+namespace Producer;
+
+internal sealed class ProducerWorker(IServiceProvider serviceProvider, ILogger<ProducerWorker> logger) : BackgroundService
+{
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        return Task.Factory.StartNew(async() =>
```

Comment:
With the `PeriodicTimer` this is not needed, as for the first time `await timer.WaitForNextTickAsync` is awaited it yields from the current task.
So in essence you create a new thread (due to LongRunning) then exit that thread at the await.

For the `BackgroundService` you can just use `await Task.Yield();` at method entry and avoid the `Task.Factory.StartNew`.

Reply:
Done. Thanks!